--- libisoburn-1.5.4/configure.ac.orig	2021-01-30 15:17:47.000000000 +0100
+++ libisoburn-1.5.4/configure.ac	2021-03-07 07:49:01.590337842 +0100
@@ -167,7 +167,7 @@
 
     else
 dnl The empty yes case obviously causes -lreadline to be linked
-        AC_CHECK_HEADER(readline/readline.h, AC_CHECK_LIB(readline, readline, , READLINE_DEF= ), READLINE_DEF= )
+        AC_CHECK_HEADER(readline/readline.h, AC_CHECK_LIB(readline, readline, READLINE_LIBS=-lreadline, READLINE_DEF= ), READLINE_DEF= )
 dnl The X= in the yes case prevents that -lreadline gets linked twice
         AC_CHECK_HEADER(readline/history.h, AC_CHECK_LIB(readline, add_history, X= , READLINE_DEF= ), READLINE_DEF= )
     fi
@@ -196,6 +196,7 @@
     fi
 fi
 AC_SUBST(READLINE_DEF)
+AC_SUBST(READLINE_LIBS)
 
 
 dnl ts A90329
--- libisoburn-1.5.4/Makefile.am.orig	2021-03-07 07:47:18.288379575 +0100
+++ libisoburn-1.5.4/Makefile.am	2021-03-07 07:49:36.902146676 +0100
@@ -118,7 +118,7 @@
 #                          $(XORRISO_DVD_OBS_64K)
 
 xorriso_xorriso_LDADD = libisoburn/libisoburn.la -lisofs -lburn \
-		$(THREAD_LIBS) $(LIBBURN_ARCH_LIBS)
+		$(THREAD_LIBS) $(LIBBURN_ARCH_LIBS) $(READLINE_LIBS)
 
 xorriso_xorriso_SOURCES = \
 	xorriso/xorriso.h \
