--- libisoburn-0.6.4/configure.ac.orig	2010-10-26 12:19:52.000000000 +0200
+++ libisoburn-0.6.4/configure.ac	2010-11-11 15:41:46.855097423 +0100
@@ -129,7 +129,7 @@
 
     else
 dnl The empty yes case obviously causes -lreadline to be linked
-        AC_CHECK_HEADER(readline/readline.h, AC_CHECK_LIB(readline, readline, , READLINE_DEF= ), READLINE_DEF= )
+        AC_CHECK_HEADER(readline/readline.h, AC_CHECK_LIB(readline, readline, READLINE_LIBS=-lreadline, READLINE_DEF= ), READLINE_DEF= )
 dnl The X= in the yes case prevents that -lreadline gets linked twice
         AC_CHECK_HEADER(readline/history.h, AC_CHECK_LIB(readline, add_history, X= , READLINE_DEF= ), READLINE_DEF= )
     fi
@@ -137,6 +137,7 @@
     READLINE_DEF=
 fi
 AC_SUBST(READLINE_DEF)
+AC_SUBST(READLINE_LIBS)
 
 dnl ts A90329
 dnl ACL and xattr do not need to be enabled in libisoburn or xorriso source
--- libisoburn-0.3.8/Makefile.am.orig	2009-04-18 13:42:40.000000000 +0200
+++ libisoburn-0.3.8/Makefile.am	2009-06-09 18:23:38.075512091 +0200
@@ -55,7 +55,7 @@
                          $(EXTF_DEF) $(EXTF_SUID_DEF) $(ZLIB_DEF)
 
 xorriso_xorriso_LDADD = libisoburn/libisoburn.la -lisofs -lburn \
-		$(THREAD_LIBS) $(LIBBURN_ARCH_LIBS)
+		$(THREAD_LIBS) $(LIBBURN_ARCH_LIBS) $(READLINE_LIBS)
 
 xorriso_xorriso_SOURCES = \
 	xorriso/xorriso.h \
